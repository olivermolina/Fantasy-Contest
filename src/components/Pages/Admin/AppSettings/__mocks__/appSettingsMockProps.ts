export const appSettingsMockProps = {
  appSettings: {
    BONUS_CREDIT_FREE_ENTRY_EQUIVALENT: '25',
    STAKE_TYPE_FREE_ENTRY: 'ALL_IN,INSURED',
    NUMBER_OF_PLAYERS_FREE_ENTRY: '2,3,4',
    MAX_MATCH_DEPOSIT_AMOUNT: '100',
    DEPOSIT_AMOUNT_OPTIONS: '10,25,50,75,100,200',
    RELOAD_BONUS_TYPE: 'FLAT',
    RELOAD_BONUS_AMOUNT: '0',
    MAX_RETENTION_BONUS: '50',
    RETENTION_BONUS_WEEKLY_CHANCE: '1',
    RETENTION_BONUS_MATCH_MULTIPLIER: '1',
    REFERRAL_CREDIT_AMOUNT: '25',
    REFERRAL_CUSTOM_TEXT:
      'Refer a friend from the link or code in your profile and get $25 in bonus credit!!',
    MIN_BET_AMOUNT: '1',
    MAX_BET_AMOUNT: '50',
    MAX_MARKET_ODDS: '150',
    MIN_MARKET_ODDS: '-150',
    initial: false,
    CHALLENGE_PROMO_MESSAGE:
      'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
    REPEAT_ENTRIES_LIMIT: '1',
  },
  onSubmit: () => console.log('onSubmit'),
};
