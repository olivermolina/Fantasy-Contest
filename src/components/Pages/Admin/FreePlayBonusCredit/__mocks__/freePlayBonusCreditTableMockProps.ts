import { PaymentStatusCode } from '~/constants/PaymentStatusCode';
import { TransactionType } from '@prisma/client';

export const freePlayBonusCreditTableMockProps = {
  onSubmitDeletePlayFreeCredit: () => console.log('delete'),
  data: [
    {
      id: '1',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.COMPLETE,
      type: TransactionType.CREDIT,
    },
    {
      id: '2',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.CANCELLED,
      type: TransactionType.DEBIT,
    },
  ],
  totalBonusCreditBalance: 0,
};
