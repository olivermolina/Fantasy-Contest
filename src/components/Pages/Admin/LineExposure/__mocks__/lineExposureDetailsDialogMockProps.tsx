import { lineExposureMockProps } from './lineExposureMockProps';

export const lineExposureDetailsDialogMockProps = {
  row: lineExposureMockProps.data[0],
  open: false,
  handleClose: () => console.log(close),
};
