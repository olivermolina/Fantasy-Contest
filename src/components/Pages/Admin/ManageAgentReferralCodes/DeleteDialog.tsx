import * as React from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';

interface Props {
  /**
   * Callback function to delete referral code
   * @param referralCode
   */
  confirmDelete: (referralCode: string) => void;
  /**
   * Callback function to close delete referral code dialog
   */
  closeDialog: () => void;
  /**
   * Boolean to show delete referral code dialog
   */
  open: boolean;
  /**
   * Referral code to delete
   */
  referral: string;
}

export default function DeleteDialog(props: Props) {
  const { closeDialog, open, confirmDelete, referral } = props;

  return (
    <Dialog
      open={open}
      onClose={closeDialog}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title">
        {'Delete Referral Code?'}
      </DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description">
          Are you sure you want to delete{' '}
          <span className={'font-semibold'}>{`'${referral}'`}</span> referral
          code?
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={closeDialog}>Cancel</Button>
        <Button
          onClick={() => confirmDelete(referral)}
          variant={'outlined'}
          autoFocus
        >
          Confirm
        </Button>
      </DialogActions>
    </Dialog>
  );
}
