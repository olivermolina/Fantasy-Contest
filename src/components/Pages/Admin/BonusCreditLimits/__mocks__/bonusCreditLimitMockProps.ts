import { contestCategoriesMock } from '~/components/Pages/Admin/ManageFreeSquarePromotion/__mocks__/contestCategoriesMock';

export const bonusCreditLimitMockProps = {
  onSubmit: () => console.log('submit'),
  contestCategories: contestCategoriesMock,
  data: {
    numberOfPlayers: ['2', '3', '4'],
    bonusCreditFreeEntryEquivalent: 25,
    stakeType: ['INSURED', 'ALL_IN'],
  },
};
