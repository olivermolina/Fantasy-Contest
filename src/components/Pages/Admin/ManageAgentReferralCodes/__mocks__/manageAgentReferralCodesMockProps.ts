import { usersMock } from '~/components/Admin/Management/__mocks__/usersMocks';

export const manageAgentReferralCodesMockProps = {
  onSubmit: () => alert('test'),
  users: usersMock,
  isLoading: false,
  deleteReferralCode: () => alert('delete'),
};
