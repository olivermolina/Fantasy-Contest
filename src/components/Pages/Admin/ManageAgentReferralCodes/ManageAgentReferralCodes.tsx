import React, { useMemo, useState } from 'react';
import UserAutocomplete from '~/components/Admin/Management/UserAutocomplete';
import { useForm } from 'react-hook-form';
import { ManagementBaseProps } from '~/components/Admin/Management/Management';
import { yupResolver } from '@hookform/resolvers/yup';
import * as Yup from 'yup';
import { ReferralCode, User, UserType } from '@prisma/client';
import {
  Button,
  Chip,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Skeleton,
  TextField,
} from '@mui/material';
import DeleteIcon from '@mui/icons-material/Delete';
import AddIcon from '@mui/icons-material/Add';
import DeleteDialog from './DeleteDialog';

const InputValidationSchema = Yup.object().shape({
  user: Yup.object()
    .typeError('Please select user')
    .required('Please select user'),
  referral: Yup.string().required('Referral code is required.'),
});

type UserWithReferralCode = User & { referralCodes: ReferralCode[] };

export interface ManageUserAgentInputs {
  user: User | string;
  referral: string;
}

interface ManageUserAgentProps extends ManagementBaseProps {
  /**
   * Submit form function
   */
  onSubmit: (inputs: ManageUserAgentInputs) => void;
  /**
   * Callback function to delete referral code
   * @param referralCodeId
   */
  deleteReferralCode: (userId: string, referralCode: string) => void;
}

export default function ManageAgentReferralCodes(props: ManageUserAgentProps) {
  const [, setUserId] = useState<string | undefined>();
  const [selectedReferral, setSelectedReferralCode] = useState('');
  const { users, isLoading, onSubmit, deleteReferralCode } = props;

  const [open, setOpen] = React.useState(false);
  const [openDeleteDialog, setOpenDeleteDialog] = React.useState(false);
  const openAddReferralCodeDialog = () => {
    setOpen(true);
  };
  const closeAddReferralCodeDialog = () => {
    setValue('referral', '');
    setOpen(false);
  };
  const openDeleteReferralCodeDialog = (referralCode: string) => {
    setSelectedReferralCode(referralCode);
    setOpenDeleteDialog(true);
  };
  const closeDeleteReferralCodeDialog = () => {
    setOpenDeleteDialog(false);
    setSelectedReferralCode('');
  };

  const {
    control,
    handleSubmit,
    register,
    formState: { errors },
    watch,
    setValue,
  } = useForm<ManageUserAgentInputs>({
    resolver: yupResolver(InputValidationSchema),
    mode: 'all',
  });

  const addNewReferralCode = () => {
    handleSubmit(onSubmit)();
    closeAddReferralCodeDialog();
  };
  const user = watch('user') as UserWithReferralCode;

  const agentReferralCodes = useMemo(
    () =>
      users
        ?.find((agent) => agent.id === user?.id)
        ?.referralCodes.map((referral) => referral.code),
    [user, users],
  );

  const confirmDeleteReferralCode = (referralCode: string) => {
    deleteReferralCode(user.id, referralCode);
    closeDeleteReferralCodeDialog();
  };

  return (
    <form>
      <div className={'flex flex-col gap-4 p-4 shadow-lg'}>
        <p className={'text-lg font-semibold'}>Agent Referral Codes</p>
        <div className={'flex flex-col gap-4'}>
          <UserAutocomplete
            control={control}
            users={users.filter((user) => user.type === UserType.AGENT)}
            isLoading={isLoading}
            setSelectedUserId={setUserId}
            label={'Select Agent'}
          />
        </div>

        {isLoading ? (
          <div className={'flex flex-wrap gap-2'}>
            <Skeleton variant="rounded" width={45} height={30} />
            <Skeleton variant="rounded" width={60} height={30} />
            <Skeleton variant="rounded" width={50} height={30} />
          </div>
        ) : (
          <div className={'flex flex-wrap gap-2'}>
            {user && <Chip label={user.username} variant="outlined" />}
            {agentReferralCodes?.map((referralCode) => (
              <Chip
                key={referralCode}
                label={referralCode}
                onDelete={() => openDeleteReferralCodeDialog(referralCode)}
                deleteIcon={<DeleteIcon />}
                variant="outlined"
              />
            ))}
            {user && (
              <Chip
                label={<AddIcon />}
                onClick={openAddReferralCodeDialog}
                variant="outlined"
              />
            )}
          </div>
        )}
      </div>
      <Dialog open={open} onClose={closeAddReferralCodeDialog}>
        <DialogTitle>Add New Referral Code</DialogTitle>
        <DialogContent>
          <TextField
            autoFocus
            margin="dense"
            id="name"
            label="Referral Code"
            fullWidth
            variant="standard"
            {...register('referral')}
            error={!!errors?.referral}
            helperText={errors?.referral?.message}
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={closeAddReferralCodeDialog} variant={'outlined'}>
            Cancel
          </Button>
          <Button onClick={addNewReferralCode} variant={'contained'}>
            Add
          </Button>
        </DialogActions>
      </Dialog>
      <DeleteDialog
        open={openDeleteDialog}
        confirmDelete={confirmDeleteReferralCode}
        referral={selectedReferral}
        closeDialog={closeDeleteReferralCodeDialog}
      />
    </form>
  );
}
