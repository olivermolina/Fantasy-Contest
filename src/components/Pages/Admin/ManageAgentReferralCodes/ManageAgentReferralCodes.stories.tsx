import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import ManageAgentReferralCodes from './ManageAgentReferralCodes';
import { manageAgentReferralCodesMockProps } from '~/components/Pages/Admin/ManageAgentReferralCodes/__mocks__/manageAgentReferralCodesMockProps';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Lockspread/Admin/Management/AgentReferralCodes',
  component: ManageAgentReferralCodes,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof ManageAgentReferralCodes>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof ManageAgentReferralCodes> = (args) => (
  <ManageAgentReferralCodes {...args} />
);

export const Default = Template.bind({});
Default.args = manageAgentReferralCodesMockProps;

export const WithLoading = Template.bind({});
WithLoading.args = {
  ...manageAgentReferralCodesMockProps,
  isLoading: true,
};
