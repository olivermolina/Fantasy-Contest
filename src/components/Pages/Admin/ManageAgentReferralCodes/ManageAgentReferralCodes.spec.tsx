import { render } from '@testing-library/react';
import ManageAgentReferralCodes from './ManageAgentReferralCodes';
import { manageAgentReferralCodesMockProps } from './__mocks__/manageAgentReferralCodesMockProps';

describe('ManageAgentReferralCodes', () => {
  it('should render correctly', () => {
    const { container } = render(
      <ManageAgentReferralCodes {...manageAgentReferralCodesMockProps} />,
    );
    expect(container).toMatchSnapshot();
  });
});
