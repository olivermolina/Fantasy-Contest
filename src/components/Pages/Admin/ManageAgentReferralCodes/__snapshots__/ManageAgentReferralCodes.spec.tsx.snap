// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageAgentReferralCodes should render correctly 1`] = `
<div>
  <form>
    <div
      class="flex flex-col gap-4 p-4 shadow-lg"
    >
      <p
        class="text-lg font-semibold"
      >
        Agent Referral Codes
      </p>
      <div
        class="flex flex-col gap-4"
      >
        <div
          class="MuiAutocomplete-root MuiAutocomplete-fullWidth css-qm9mjt-MuiAutocomplete-root"
          name="user"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="userId"
              id="userId-label"
            >
              Select Agent
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                id="userId"
                role="combobox"
                spellcheck="false"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Select Agent
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-wrap gap-2"
      />
    </div>
  </form>
</div>
`;
